### DISCUSSION TITLE
8k-SG01LP1-EU entities

### DISCUSSION THREAD

**Discussion by nagy592** [StephanJoubert/home_assistant_solarman#505] [Q&A]
I have the integration up and running awhile now, and it seems working fine. My issue is, I can't find one entity which is related to the essential load (called UPS on Deye cloud dashboard). I tried various .yaml options, the the best is definitely the deye_hybrid.ymal, but no luck with the UPS. I only have a combined value for the Load+UPS load. Is there anyone have had luck with this?
![Deye powerflow](https://github.com/StephanJoubert/home_assistant_solarman/assets/4986146/e1d82b98-8c07-45c4-b989-0164c729804e)

**Comment by blindeslakkie** [comment]
Hi
I'm also running the deye_hybrid.ymal config.
I'm looking for the essential and non-essential loads, but can not get it.
Is there any way to integrate this?

**Comment by anderslew** [comment]
I have this running for the 12k model. It is the registers 640 to 643 (converted to HEX in the below config). Had to find them in a document that was half translated ;-)

I can post my full config if relevant. You still need to change some parameters in the top to actually include the registers.

    - name: "Total UPS Power"
      class: "power"
      state_class: "measurement"      
      uom: "W"
      scale: 1
      rule: 1
      registers: [0x0283]
      icon: 'mdi:lightning-bolt-outline'

    - name: "UPS L1 Power"
      class: "power"
      state_class: "measurement"      
      uom: "W"
      scale: 1
      rule: 1
      registers: [0x0280]
      icon: 'mdi:lightning-bolt-outline'

    - name: "UPS L2 Power"
      class: "power"
      state_class: "measurement"      
      uom: "W"
      scale: 1
      rule: 1
      registers: [0x0281]
      icon: 'mdi:lightning-bolt-outline'
      
    - name: "UPS L3 Power"
      class: "power"
      state_class: "measurement"      
      uom: "W"
      scale: 1
      rule: 1
      registers: [0x0282]
      icon: 'mdi:lightning-bolt-outline'

**Reply by anderslew** [reply]
@nagy592 like this :-)
<img width="177" alt="image" src="https://github.com/StephanJoubert/home_assistant_solarman/assets/79974408/57ce1c2e-4867-4156-83fa-678b993583f2">

**Reply by nagy592** [reply]
Is it a 3 phase unit?

**Reply by anderslew** [reply]
Yes

**Reply by nagy592** [reply]
Which inverter definition you use? I found deye_hybrid.yaml the most relevant.

**Reply by anderslew** [reply]
I used deye_sg04lp3.yaml as a building block (since mine is the 3 phase unit). And I'm not sure if the registers are the same for your unit. But maybe worth a try.

A bit more background. With my installation I have two car chargers sitting alone on Grid (with CT clamps in place right after power meter).
The whole house then runs directly on the Backup load (I have a manual switch so that I can bypass the inverter in case of failure or similar).

If I then need to charge from excess solar power, I set the inverter to "Zero Export To CT". And take PV Power minus Total UPS Power to know what to aim for ;-) I also start charging when the house batter is around 75%, and use the inverter to balance import/export to 0, even when charging. Works excellent :-)

If I need to charge the cars in Winter, I then set the inverter to "Zero Export To Load", and then there is not risk of draining the house batteries.

**Reply by nagy592** [reply]
Yeah, I don't have the registers you used for "UPS"... must be something different.
Anyway, my system is almost identical to yours, apart from I have large consumers on the "load" side (oven, microwave, hob, HWC, Heat-pump) and the rest on the "UPS", with a change over switch to grid... To avoid discharging the house battery, I have an automation, to turn off the charger, I have a semi dumb charger.
![EV stop](https://github.com/StephanJoubert/home_assistant_solarman/assets/4986146/2974cae9-beed-4075-b4f3-0785b0f6fed8)

**Comment by ks0002544** [comment]
do you mean like this?

<img width="389" alt="Deye_Solarman" src="https://github.com/StephanJoubert/home_assistant_solarman/assets/125402551/27ba784a-3817-4f1a-b1f1-997446663845">

**Reply by nagy592** [reply]
Yes. To be fair no problem to setup the card, but I couldn't allocate the relevant data in the Solarman list to insert to it.

**Reply by ks0002544** [reply]
Do you use CT on grid line?

**Reply by nagy592** [reply]
Yes I do.

**Reply by ks0002544** [reply]
check this
type: custom:sunsynk-power-flow-card
cardstyle: full
panel_mode: false
large_font: true
title: DEYE / TESLA
title_colour: grey
title_size: 24px
show_solar: true
show_battery: true
show_grid: true
decimal_places: 2
dynamic_line_width: true
min_line_width: 2
max_line_width: 5
inverter:
  modern: false
  colour: grey
  autarky: power
  auto_scale: true
battery:
  energy: 6144
  colour: pink
  show_daily: true
  animation_speed: 5
  max_power: 4500
  invert_power: false
  show_absolute: null
  auto_scale: false
  hide_soc: false
  dynamic_colour: true
  linear_gradient: true
  animate: true
  path_threshold: 100
  shutdown_soc: 30
solar:
  colour: yellow
  show_daily: true
  mppts: 2
  animation_speed: 5
  max_power: 5750
  pv1_name: PV1
  pv1_max_power: 2875
  pv2_name: PV2
  pv2_max_power: 2875
  auto_scale: true
  display_mode: 1
  dynamic_colour: true
  efficiency: 3
  off_threshold: 0
load:
  colour: red
  show_daily: true
  animation_speed: 2
  max_power: 6000
  auto_scale: false
  load3_name: MeterOUT
  load3_icon: mdi:meter-electric
  load2_name: Kitchen
  load2_icon: mdi:glass-stange
  load1_name: MeterIN
  load1_icon: mdi:meter-electric-outline
  load4_name: WaterSt
  load4_icon: mdi:water-boiler
  additional_loads: 4
  dynamic_colour: true
  off_threshold: 0
grid:
  colour: orange
  export_colour: brown
  grid_off_colour: brown
  show_daily_buy: true
  show_daily_sell: null
  no_grid_colour: white
  show_nonessential: true
  additional_loads: 2
  load1_name: A-----TE
  load1_icon: mdi:ev-plug-tesla
  load2_name: SLA------%
  load2_icon: mdi:ev-plug-tesla
  invert_grid: false
  animation_speed: 5
  dynamic_colour: true
  max_power: 6000
  auto_scale: false
  off_threshold: 0
entities:
  essential_load2: sensor.sp3s_16a_kitchen_power
  essential_load1: sensor.wifi_din_rail_switch_with_metering_power
  essential_load3: sensor.dacha_umnyi_schetchik_phase_a_power
  essential_load4: sensor.vodnaia_stantsiia_power
  non_essential_load1: number.sergemobil_x_charging_amps
  non_essential_load2: sensor.sergemobil_x_battery
  pv1_power_186: sensor.deye_pv1_power
  pv2_power_187: sensor.deye_pv2_power
  day_pv_energy_108: sensor.deye_daily_production
  pv1_voltage_109: sensor.deye_pv1_voltage
  pv1_current_110: sensor.deye_pv1_current
  pv2_voltage_111: sensor.deye_pv2_voltage
  pv2_current_112: sensor.deye_pv2_current
  battery_voltage_183: sensor.deye_battery_voltage
  battery_soc_184: sensor.deye_battery_soc
  battery_power_190: sensor.deye_battery_power
  battery_current_191: sensor.deye_battery_current
  day_battery_charge_70: sensor.deye_daily_battery_charge
  day_battery_discharge_71: sensor.deye_daily_battery_discharge
  grid_power_169: sensor.deye_internal_ct_l1_power
  inverter_power_175: sensor.deye_total_power
  /*inverter_current_164: sensor.sergemobil_x_time_charge_complete
  essential_power: none
  nonessential_power: none
  day_grid_import_76: sensor.deye_daily_energy_bought
  day_grid_export_77: sensor.deye_daily_energy_sold
  grid_ct_power_172: sensor.deye_external_ct_l1_power
  grid_connected_status_194: sensor.deye_grid_connected_status
  day_load_energy_84: sensor.deye_daily_load_consumption
  inverter_status_59: sensor.deye_running_status
  battery_status: none

**Reply by ks0002544** [reply]
these entities are from Solarman integration

**Reply by ks0002544** [reply]
have you got 3 phase invertor?

**Reply by nagy592** [reply]
No, my one is single phase.

**Reply by nagy592** [reply]
type: custom:sunsynk-power-flow-card
cardstyle: full
panel_mode: false
title: Deye 8kW
title_colour: blue
show_solar: true
inverter:
  modern: true
  colour:
    - 73
    - 228
    - 218
  auto_scale: true
  model: DEYE 8k
  autarky: energy
battery:
  energy: 25000
  shutdown_soc: 5
  max_power: 8000
  full_capacity: 100
  empty_capacity: 8
  show_daily: true
  inverter_power: true
  colour: green
  auto_scale: true
  dynamic_colour: true
  linear_gradient: true
  animation_speed: 6
  show_remaining_energy: true
  hide_soc: false
solar:
  show_daily: false
  mppts: 2
  auto_scale: true
  display_mode: 2
  dynamic_colour: true
  max_power: 8800
  animation_speed: 5
  pv1_max_power: 5280
  efficiency: 1
  pv2_max_power: 3520
load:
  show_daily: true
  show_daily_aux: true
  show_aux: false
  aux_load_icon: mdi:ev-station
  aux_colour: green
  aux_off_colour: red
  additional_loads: 0
  load1_name: Daikin
  load2_name: EV
  load1_icon: mdi:heat-wave
  load2_icon: mdi:ev-station
  auto_scale: true
  dynamic_colour: true
  dynamic_icon: true
  aux_name: Heating
  invert_aux: false
  aux_type: mdi:heat-wave
  aux_dynamic_colour: true
  aux_loads: 2
grid:
  colour: '#c254aa'
  export_colour: green
  import_colour: red
  no_grid_colour: blue
  show_daily_buy: true
  show_daily_sell: true
  show_nonessential: true
  nonessential_name: Grid supply
  invert_grid: false
  nonessential_icon: mdi:factory
  load1_name: EV
  load1_icon: mdi:ev-station
  load2_name: Daikin
  load2_icon: mdi:heat-wave
  additional_loads: 2
  energy_cost_decimals: 2
entities:
  use_timer_248: binary_sensors:electricity_night_rates
  priority_load_243: switch.solarman_toggle_priority_load
  inverter_voltage_154: sensor.solarman_grid_voltage_l1
  load_frequency_192: sensor.solarman_grid_frequency
  inverter_current_164: sensor.solarman_current_l1
  inverter_power_175: sensor.solarman_inverter_l1_power
  grid_connected_status_194: sensor.solarman_grid_connected_status
  inverter_status_59: sensor.solarman_status_connection
  day_battery_charge_70: sensor.solarman_daily_battery_charge
  day_battery_discharge_71: sensor.solarman_daily_battery_discharge
  battery_voltage_183: sensor.solarman_battery_voltage
  battery_soc_184: sensor.solarman_battery_soc
  battery_power_190: sensor.solarman_battery_power
  battery_current_191: sensor.solarman_battery_current
  grid_power_169: sensor.solarman_external_ct_l1_voltage
  day_grid_import_76: sensor.solarman_daily_energy_bought
  day_grid_export_77: sensor.solarman_daily_energy_sold
  day_aux_energy: sensor.duosida_accenergy2
  grid_ct_power_172: sensor.solarman_external_ct_l1_voltage
  day_load_energy_84: sensor.solarman_daily_load_consumption
  essential_power: sensor.solarman_total_load_power
  essential_load1: sensor.electricity_used
  essential_load2: sensor.duosida_power
  nonessential_power: null
  day_pv_energy_108: sensor.solarman_daily_production
  remaining_solar: sensor.energy_production_today_remaining
  pv1_power_186: sensor.solarman_pv1_power
  pv2_power_187: sensor.solarman_pv2_power
  pv1_voltage_109: sensor.solarman_pv1_voltage
  pv1_current_110: sensor.solarman_pv1_current
  pv2_voltage_111: sensor.solarman_pv2_voltage
  pv2_current_112: sensor.solarman_pv2_current
  grid_ct_power_L2: sensor.solarman_total_grid_power
  non_essential_load1: sensor.duosida_power
  non_essential_load2: sensor.electricity_used
  aux_load1: sensor.electricity_used
  aux_load2: sensor.duosida_power
large_font: true
show_battery: true
show_grid: true
dynamic_line_width: true
max_line_width: 20
min_line_width: 2
decimal_places: 3

**Reply by ks0002544** [reply]
ok then use these entities:

grid_power_169: sensor.deye_internal_ct_l1_power

grid_ct_power_172: sensor.deye_external_ct_l1_power

**Comment by nagy592** [comment]
I did the same thing what you did, with additional data, but that is not from Solarman.

_"essential_load2: sensor.sp3s_16a_kitchen_power
essential_load1: sensor.wifi_din_rail_switch_with_metering_power
essential_load3: sensor.dacha_umnyi_schetchik_phase_a_power
essential_load4: sensor.vodnaia_stantsiia_power
non_essential_load1: number.sergemobil_x_charging_amps
non_essential_load2: sensor.sergemobil_x_battery"_

**Reply by nagy592** [reply]
non_essential_load1: **sensor.duosida_**power
non_essential_load2: **sensor.electricity**_used

**Reply by ks0002544** [reply]
these entities also not from Solarman
"essential_load2: sensor.sp3s_16a_kitchen_power
essential_load1: sensor.wifi_din_rail_switch_with_metering_power
essential_load3: sensor.dacha_umnyi_schetchik_phase_a_power
essential_load4: sensor.vodnaia_stantsiia_power
non_essential_load1: number.sergemobil_x_charging_amps
non_essential_load2: sensor.sergemobil_x_battery"

this is for charge control of tesla

**Comment by ks0002544** [comment]
these entities need exactly for non essential load
grid_power_169: sensor.deye_internal_ct_l1_power

grid_ct_power_172: sensor.deye_external_ct_l1_power

**Reply by nagy592** [reply]
![CTs](https://github.com/StephanJoubert/home_assistant_solarman/assets/4986146/72610d00-e4b1-4b3a-a678-83095a1d3df0)


External CT shows 0 because no use or feed to the grid.

**Reply by ks0002544** [reply]
are you selling some solar power to the grid?

**Reply by nagy592** [reply]
Are you completely offgrid?

**Reply by nagy592** [reply]
Occasionally. Summer time only, not now.

**Reply by nagy592** [reply]
Trying to minimize the "damage" by selling it.

**Reply by ks0002544** [reply]
no, sometimes electicity company switchs off the grid

**Comment by ks0002544** [comment]
if now not selling then you need to reinstall CT correct

**Reply by nagy592** [reply]
![zero to CT](https://github.com/StephanJoubert/home_assistant_solarman/assets/4986146/c7aaf7d5-118e-4e26-ba08-fae842637dcd)
Where exactly you suggest compare to this on the picture?

**Reply by ks0002544** [reply]
Did you see arrow on the CT unit?

**Reply by nagy592** [reply]
Yes. It works as it should... negative when selling, positive when buying.

**Reply by nagy592** [reply]
![CT joke](https://github.com/StephanJoubert/home_assistant_solarman/assets/4986146/dcf6b09e-2cdb-4d1a-9b84-eab6648e7e5c)
But, to be fair, after this, which is in the manual, I don't think the arrow is relevant at all

**Reply by ks0002544** [reply]
if vice versa then CT must be turned the arrow to the other side

the arrow is very relevant at all
